modules/library/render/src/main/java/org/geotools/renderer/style/ImageGraphicFactory.java

```diff
@@ -65,6 +66,10 @@ public Icon getIcon(Feature feature, Expression url, String format, int size) {
             throw new IllegalArgumentException(
                     "The provided expression cannot be evaluated to a URL");
 
+        // validate the icon can actually be fetched, it may go to a random
+        // local filesystem location, or a remote server
+        URLCheckers.confirm(location);
```

Comment:
This not work out as I was anticipating from the proposal; for some reason I thought we would have more fine grain control over individual URLCheckers.

As an example just because schemas.opengis.net is allowed for entity resolution does not mean it should be available here for use as external graphic ...

Reply:
The proposal does not contain anything like that.... nor did the original proposal it replaces. A remote host is safe to contact, or not, regardless of the context... (mind, this is simple security, it's not "does it make sense to contact remote host in this execution context")